src/ASM/LR/ASMu2D.C

```diff
@@ -2764,12 +2764,15 @@ const RealArray& ASMu2D::InterfaceChecker::getIntersections (int iel, int edge,
 bool ASMu2D::refine (const LR::RefineData& prm, Vectors& sol)
 {
   bool ok = this->ASMLRSpline::refine(prm,sol);
-  if (!ok || !this->separateProjectionBasis() ||
-      prm.elements.size() + prm.errors.size() == 0)
+
+  // check if refinement was actually done
+  if (prm.elements.size() + prm.errors.size() == 0 || !ok)
     return ok;
```

Comment:
To make it even more readable, we can drop the `ok` variable entirely:
```
if (!this->ASMLRSpline::refine(prm,sol))
  return false;

// check if refinement was actually done
if (prm.elements.empty() && prm.errors.empty())
  return true;
```

Reply:
done.